sql-plugin/src/main/340+/scala/com/nvidia/spark/rapids/shims/Spark340PlusShims.scala

```diff
@@ -76,7 +78,59 @@ trait Spark340PlusShims extends Spark331PlusShims {
         (a, conf, p, r) => new UnaryExprMeta[Empty2Null](a, conf, p, r) {
           override def convertToGpu(child: Expression): GpuExpression = GpuEmpty2Null(child)
         }
-      )
+      ),
+      GpuOverrides.expr[ElementAt](
+        "Returns element of array at given(1-based) index in value if column is array. " +
+          "Returns value for the given key in value if column is map.",
+        ExprChecks.binaryProject(
+          (TypeSig.commonCudfTypes + TypeSig.ARRAY + TypeSig.STRUCT + TypeSig.NULL +
+            TypeSig.DECIMAL_128 + TypeSig.MAP + TypeSig.BINARY).nested(), TypeSig.all,
+          ("array/map", TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.ARRAY +
+            TypeSig.STRUCT + TypeSig.NULL + TypeSig.DECIMAL_128 + TypeSig.MAP + TypeSig.BINARY) +
+            TypeSig.MAP.nested(TypeSig.commonCudfTypes + TypeSig.ARRAY + TypeSig.STRUCT +
+              TypeSig.NULL + TypeSig.DECIMAL_128 + TypeSig.MAP + TypeSig.BINARY)
+              .withPsNote(TypeEnum.MAP ,"If it's map, only primitive key types are supported."),
+            TypeSig.ARRAY.nested(TypeSig.all) + TypeSig.MAP.nested(TypeSig.all)),
+          ("index/key", (TypeSig.commonCudfTypes + TypeSig.DECIMAL_128)
+            .withPsNote(
+              Seq(TypeEnum.BOOLEAN, TypeEnum.BYTE, TypeEnum.SHORT, TypeEnum.LONG,
+                TypeEnum.FLOAT, TypeEnum.DOUBLE, TypeEnum.DATE, TypeEnum.TIMESTAMP,
+                TypeEnum.STRING, TypeEnum.DECIMAL), "Unsupported as array index."),
+            TypeSig.all)),
+        (in, conf, p, r) => new BinaryExprMeta[ElementAt](in, conf, p, r) {
+          override def tagExprForGpu(): Unit = {
```

Comment:
Better to create an abstract class as the base to cover the common code for this meta class, to avoid copying this long method `tagExprForGpu` here.